Add FileUtils#cp

Dreauw opened the pull request:
Adds a method to copy a file or a list of files in the class FileUtils.
And the associated tests in spec/std/file_utils_spec.cr


:dolphin:

jhass commented on src/file_utils.cr:
```diff
@@ -24,4 +24,29 @@ module FileUtils
     end
   end
+
+  # Copies a file content *src* to *dest*. If dest is a directory, copies *src* to *dest/src*
```
`file contents of *src_path*` might be less confusing. Also mentioning that the basename is taken in the second case might be worth it.

Dreauw commented:
Yep, the documentation was confusing so I rewrote it.

Should be better now :)

jhass commented on src/file_utils.cr:
```diff
@@ -24,4 +24,31 @@ module FileUtils
+  def cp(srcs : Enumerable(String), dest : String)
+    Dir.mkdir(dest) rescue nil
+    srcs.each do |src|
+      cp(src, dest + File::SEPARATOR)
```
Actually why append `File::SEPARATOR`? I think `raise ArgumentError.new("...") unless Dir.exists?(dest)` would be better :)

Dreauw commented:
Yep, the "File::SEPARATOR" is totally useless, good catch !

Good idea about raising an error, I dunno why I didn't thought of that in the first place (and it's the behavior used in ruby actually).
I changed the code and added a test case :)

jhass commented:
LGTM :+1:

kirbyfan64 commented:
Could you make this also copy the basic attributes (e.g. mode) like [Python's `shutil.copy`](https://docs.python.org/3/library/shutil.html#shutil.copy) does?

Dreauw commented:
@kirbyfan64 That should be already the case. The permissions bits are copied too.

ysbaddaden commented:
Thanks! FileUtils is desperately in need for more :heart:
